api/config/services.yaml

```
@@ -33,3 +33,5 @@ services:
     App\Handler\BookHandler:
         arguments:
             $imgflipClient: '@csa_guzzle.client.imgflip'
+
+    App\Repository\TopBook\TopBookDataInterface: '@App\Repository\TopBook\TopBookCachedDataRepository'
```

```suggestion
    App\Repository\TopBook\TopBookDataInterface: '@App\Repository\TopBook\TopBookDataRepository'
```
The service decoration will handle the rest.

The `TopBookCachedDataRepository` service should decorate `TopBookDataRepository`